doc/contributor/howto-guide-setup-cmake-environment.md

```diff
@@ -0,0 +1,109 @@
+# How-to Guide: setting for CMake-based builds
+
+This document describes how to setup your workstation to build the Google Cloud
+C++ client libraries using CMake. The intended audience is developers of the
+client libraries that want to verify their changes will work with CMake and/or
+prefer to use CMake for whatever reason.
+
+* Packaging maintainers or developers that prefer to install the library in a
+  fixed directory (such as `/usr/local` or `/opt`) should consult the
+  [packaging guide](/doc/packaging.md).
+* Developers wanting to use the libraries as part of a larger CMake or Bazel
+  project should consult the [quickstart guides](/README.md#quickstart) for the library
+  or libraries they want to use.
+* Developers wanting to compile the library just to run some of the examples or
+  tests should read the [build and install](/README.md#building-and-installing)
+  section from the top-level README.
+* Contributors and developers to `google-cloud-cpp` wanting to work with CMake
+  should read the current document.
+
+The document assumes you are using a Linux workstation running Ubuntu, changing
+the instructions for other distributions or operating systems is left as an
+exercise to the reader. PRs to improve this document are most welcome!
+
+Unless explicitly stated, this document assumes you running these commands at
+the top-level directory of the project, as in:
+
+```console
+cd $HOME
+git clone git@github:<github-username>/google-cloud-cpp.git
+cd google-cloud-cpp
+```
+
+## Install the dependencies in `$HOME/local-cpp`.
+
+In this guide we will install the dependencies in `$HOME/local-cpp`, you can
+pick any directory where you have write permissions. We will then compile the
+project against these dependencies. The installation needs to be done every time
+the version of the dependencies is changed. Once installed you can use them for
+any build or from any other clone.
+
+Configure the super-build to install in `$HOME/local-cpp`. We recommend that
+you use Ninja for this build because it is substantially faster than Make for
+these use-cases. Note that if Ninja is not installed in your workstation you may
+need to remove the `-GNinja` flag:
+
+```console
+cmake -Hsuper -Bcmake-out/si \
+    -DGOOGLE_CLOUD_CPP_EXTERNAL_PREFIX=$HOME/local-cpp -GNinja
+```
+
+Install the dependencies:
+
+```console
+cmake --build cmake-out/si --target project-dependencies -- -j $(nproc)
```

Comment:
Is the `-j` needed? I thought that Ninja would use all available processors by default.

Reply:
Fixed